### PR TITLE
Modified the files to

### PR DESCRIPTION
**Pull request by SubhadraSahoo**
1. upgrade the cx-client-common to 2021.2.159 which is latest
2. Fixed OSA , SCA, UI visibility of OSA fields at build level
3. Code clean upand addressed the TODOs as per the comments put earlier
4. Changed the team city plugin version info in all pom.xml files

### CODE REVIEW

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxBuildProcess.java [hunk 1]
```diff
@@ -153,22 +150,16 @@ public BuildFinishedStatus call() throws Exception {
...
+                    (config.isOsaEnabled() && (ret.getOsaResults() == null || ret.getOsaResults().getException() != null)) ||
+                    (config.isAstScaEnabled() && (ret.getScaResults() == null || ret.getScaResults().getException() != null))) {
                 printScanBuildFailure(scanSummary, ret, logger);
                 return BuildFinishedStatus.FINISHED_FAILED;
             }
             return BuildFinishedStatus.FINISHED_SUCCESS;
         } catch (InterruptedException e) {
             logger.error("Interrupted exception: " + e.getMessage());
-
-            //TODO CHECK HOW TO CANCLE SCAN IN NEW COMMON
```
Comment (cx-umesh-waghode) [inline cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxBuildProcess.java:167]: cancel is performed using CxClientDelegator in Jenkins. You can test it by explicitly calling this function from a Test code while the scan is in-progress.  If you see it is working, remove the TODO and uncomment the cancelScan function on like 184

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java [hunk 2]
```diff
@@ -69,29 +68,41 @@ public static CxScanConfig resolveConfigurations(Map<String, String> buildParame
...
             if (TRUE.equals(buildParameters.get(OVERRIDE_GLOBAL_CONFIGURATIONS)))
             {
-                scannerType = "SCA".equalsIgnoreCase(buildParameters.get(DEPENDENCY_SCANNER_TYPE)) ?
-                        ScannerType.AST_SCA:ScannerType.OSA;
+            	ret.setOsaFilterPattern(buildParameters.get(OSA_FILTER_PATTERNS));
+                if("SCA".equalsIgnoreCase(buildParameters.get(DEPENDENCY_SCANNER_TYPE))) {
+                	scannerType = ScannerType.AST_SCA;
+                	ret.setAstScaConfig(getScaConfig(buildParameters, false));
+                }
+                else {
```
Comment (cx-umesh-waghode) [inline cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java:108]: As you have just else, it is same as the condition on line 81

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java [hunk 3]
```diff
@@ -69,29 +68,41 @@ public static CxScanConfig resolveConfigurations(Map<String, String> buildParame
...
-        }
 
         if (TRUE.equals(buildParameters.get(DEPENDENCY_SCAN_ENABLED)))
         {
             ScannerType scannerType;
             if (TRUE.equals(buildParameters.get(OVERRIDE_GLOBAL_CONFIGURATIONS)))
             {
-                scannerType = "SCA".equalsIgnoreCase(buildParameters.get(DEPENDENCY_SCANNER_TYPE)) ?
-                        ScannerType.AST_SCA:ScannerType.OSA;
+            	ret.setOsaFilterPattern(buildParameters.get(OSA_FILTER_PATTERNS));
```
Comment (cx-umesh-waghode) [inline cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java:103]: ThresholdEnabled is not set in "ret"

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java [hunk 4]
```diff
@@ -69,29 +68,41 @@ public static CxScanConfig resolveConfigurations(Map<String, String> buildParame
...
+                else {
+                	scannerType = ScannerType.OSA;
+                	ret.setOsaArchiveIncludePatterns(buildParameters.get(OSA_ARCHIVE_INCLUDE_PATTERNS));
+                    ret.setOsaRunInstall(TRUE.equals(buildParameters.get(OSA_INSTALL_BEFORE_SCAN)));
+                }
+                
             } else {
-                scannerType = "SCA".equalsIgnoreCase(buildParameters.get(GLOBAL_DEPENDENCY_SCANNER_TYPE)) ?
-                        ScannerType.AST_SCA:ScannerType.OSA;
+            	ret.setOsaFilterPattern(globalParameters.get(GLOBAL_DEPENDENCY_SCAN_FILTER_PATTERNS));
```
Comment (cx-umesh-waghode) [inline cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java:115]: ThresholdEnabled not set

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java [hunk 5]
```diff
@@ -105,9 +116,6 @@ public static CxScanConfig resolveConfigurations(Map<String, String> buildParame
 
         String isSynchronous = IS_SYNCHRONOUS;
         String enablePolicyViolation = PROJECT_POLICY_VIOLATION;
-        // TODO: 2/13/2020  add parameters that is common for two pages
```
Comment (cx-umesh-waghode) [inline cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java:108]: Why is there a today, We cannot commit with TODOs

File: cxplugin-server/src/main/resources/buildServerResources/editRunParams.jsp [hunk 6]
```diff
@@ -369,7 +369,7 @@ optionsBean.testConnection(cxServerUrl, cxUsername, cxPassword)}
         <td>
             <props:selectProperty name="${optionsBean.teamId}" className="longField">
```
Comment (cx-umesh-waghode) [inline cxplugin-server/src/main/resources/buildServerResources/editRunParams.jsp:465]: "longField" is that for number type values? Now we changing it to string type. check if that matters.

**Review by cx-umesh-waghode — CHANGES_REQUESTED** [review]

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxBuildProcess.java (hunk 1 above)
Comment (SubhadraSahoo) [reply cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxBuildProcess.java:167]: These are the removed lines. Have removed the TODO and unnecessary commented lines.

Comment (cx-umesh-waghode) [reply cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxBuildProcess.java:167]: ok. My bad

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java (hunk 3 above)
Comment (cx-umesh-waghode) [reply cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java:103]: I take it back. Its set below.

File: cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java (hunk 5 above)
Comment (cx-umesh-waghode) [reply cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxConfigHelper.java:108]: Removed lines

**Review by cx-umesh-waghode — APPROVED** [review]
Looks good

**Pull request merged**